src/profiles_management/helpers/kfam.py

```diff
@@ -0,0 +1,102 @@
+"""Utility module for manipulating KFAM resources."""
+
+import logging
+from typing import List
+
+from lightkube import Client
+from lightkube.generic_resource import GenericNamespacedResource, create_namespaced_resource
+from lightkube.resources.rbac_authorization_v1 import RoleBinding
+
+log = logging.getLogger(__name__)
+
+AuthorizationPolicy = create_namespaced_resource(
+    group="security.istio.io",
+    version="v1beta1",
+    kind="AuthorizationPolicy",
+    plural="authorizationpolicies",
+)
+
+
+def has_kfam_annotations(resource: GenericNamespacedResource) -> bool:
+    """Check if resource has "user" and "role" KFAM annotations."""
+    return (
+        resource.metadata
+        and resource.metadata.annotations
+        and "role" in resource.metadata.annotations
+        and "user" in resource.metadata.annotations
+    )
+
+
+def resource_is_for_profile_owner(resource: GenericNamespacedResource) -> bool:
+    """Check if the resource is for the Profile owner."""
+    if not resource.metadata:
+        return False
+
+    if not has_kfam_annotations(resource):
```

Comment:
This is called twice, once here and the other one in `list_contributor_*`. I recommend you remove it from here as there is no reason to do this check when also checking for owner.

Reply:
Naah you're right, we can remove it

Reply:
Done in https://github.com/canonical/github-profiles-automator/commit/325b7c4c6f15234554aed888ee53d61aabb24fe8 which includes the changes for https://github.com/canonical/github-profiles-automator/pull/17#discussion_r1909163178